src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php

```diff
@@ -46,9 +47,19 @@ public function logKernelException(ExceptionEvent $event)
     {
         $throwable = $event->getThrowable();
         $logLevel = null;
+
         foreach ($this->exceptionsMapping as $class => $config) {
-            if ($throwable instanceof $class && $config['log_level']) {
+            if (!$throwable instanceof $class) {
+                continue;
+            }
+            if ($config['log_level'] && null === $logLevel) {
                 $logLevel = $config['log_level'];
+            }
+            if ($config['status_code'] && $event->getThrowable() === $throwable) {
+                $throwable = new HttpException($config['status_code'], $throwable->getMessage(), $throwable);
+                $event->setThrowable($throwable);
+            }
+            if (null !== $logLevel && $event->getThrowable() !== $throwable) {
```

Comment:
The condition `$event->getThrowable() !== $throwable` won't ever catch the case where you override the exception, because line 59 overrides the `$throwable` variable.

Reply:
oops, fixed